packages/react-native/ReactCommon/jsc/JSCRuntime.cpp

```diff
@@ -398,6 +398,13 @@ JSCRuntime::JSCRuntime(JSGlobalContextRef ctx)
       stringCounter_(0)
 #endif
 {
+#ifndef NDEBUG
+#ifndef ANDROID
+  if (__builtin_available(macOS 13.3, iOS 16.4, tvOS 16.4, *)) {
```

Comment:
@acoates-ms: Does Windows use this code? Will this break MSVC?

Reply:
No react-native-windows never uses JSC. -- Having said that this code does look like its specific to apple platforms, so the #ifdef should probably be #if apple, rather than #if !android.

Reply:
> so the #ifdef should probably be #if apple, rather than #if !android.

True 👍 cc @huntie

Reply:
https://chaosinmotion.com/2021/08/02/things-to-remember-compiler-conditionals-for-macos-ios-etc/

How about TARGET_OS_MAC ?